api/src/Security/Authenticator/HttpBearerAuthenticator.php

```diff
@@ -0,0 +1,159 @@
+        $this->client = $client ?: HttpClient::create();
+        $this->logger = $logger;
+    }
+
+    public function start(Request $request, AuthenticationException $authException = null): Response
+    {
+        $response = new Response();
+        $response->headers->set('WWW-Authenticate', sprintf('Bearer realm="%s"', $this->realmName));
+        $response->setStatusCode(401);
+
+        return $response;
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return $request->headers->has('AUTHORIZATION');
+    }
+
+    public function authenticate(Request $request): Passport
+    {
+        if (!class_exists(Configuration::class)) {
+            throw new RuntimeException('HttpBearerAuthenticator requires lcobucci/jwt, please run "composer require lcobucci/jwt" to install it.');
+        }
+
+        try {
+            $configuration = Configuration::forSymmetricSigner($this->signer, $this->getKey($this->key));
+
+            // Decode token
+            $token = $configuration->parser()->parse(substr($request->headers->get('AUTHORIZATION'), 7));
+
+            // Validate token
+            $configuration->setValidationConstraints(new SignedWith($configuration->signer(), $configuration->signingKey()));
+            $configuration->validator()->assert($token, ...$configuration->validationConstraints());
+        } catch (Exception $exception) {
+            throw new AuthenticationException($exception->getMessage(), $exception->getCode(), $exception);
+        }
+
+        return new TokenPassport(
+            new TokenBadge($token, $token->claims()->get($this->payloadKey), [$this->userProvider, 'loadUserByIdentifier'])
+        );
+    }
+
+    public function createToken(Passport $passport, string $firewallName): TokenInterface
+    {
+        return new PostAuthenticationToken($passport->getUser(), $firewallName, $passport->getUser()->getRoles());
+    }
+
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $firewallName): ?Response
+    {
+        return null;
+    }
+
+    public function onAuthenticationFailure(Request $request, AuthenticationException $exception): ?Response
+    {
+        if (null !== $this->logger) {
+            $this->logger->info('Bearer authentication failed for token.', ['token' => $request->headers->get('AUTHORIZATION'), 'exception' => $exception]);
+        }
+
+        return $this->start($request, $exception);
+    }
+
+    private function getKey(string $key): Signer\Key
+    {
+        // It's an url: retrieve key from JSON (e.g.: https://keycloak:8443/auth/realms/master)
+        if (str_starts_with($key, 'http')) {
+            try {
+                return InMemory::plainText(sprintf(<<<KEY
+-----BEGIN PUBLIC KEY-----
+%s
+-----END PUBLIC KEY-----
+KEY
+                    , $this->client->request('GET', $key)->toArray()['public_key']
+                ));
+            } catch (ExceptionInterface $e) {
```

Comment:
seems a bit wide to me using ExceptionInterface, if there is a network/authorization issue during the request, or parsing issue, anything from the contract, it might lead to unexpected behaviour, and harden understand the problem.

Reply:
Code has changed. Now, if any exception occurres during the request, it'll be automatically converted to an AuthenticationException